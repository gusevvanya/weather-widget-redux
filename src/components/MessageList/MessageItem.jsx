import React from 'react';
import PropTypes from 'prop-types';

const MessageItem = ({ value, type }) => (
  <li className={type}>
    {value}
  </li>
);

MessageItem.propTypes = {
  value: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

export default MessageItem;
