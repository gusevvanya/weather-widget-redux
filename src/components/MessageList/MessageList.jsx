import React from 'react';
import PropTypes from 'prop-types';

import MessageItem from './MessageItem';

const MessageList = ({ messages, onClearMessageClick }) => {
  if (!messages) return null;

  const buttonClickHandler = (e) => {
    e.preventDefault();
    onClearMessageClick();
  };
  const messagesList = messages.map((message, index) => (
    <MessageItem
      key={index}
      value={message.value}
      type={message.type}
    />
  ));

  return (
    messages && (
      <div className="messageList">
        <ul>
          {messagesList}
        </ul>
        <button
          onClick={buttonClickHandler}
        >
          x
        </button>
      </div>
    )
  );
};

MessageList.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.object).isRequired,
  onClearMessageClick: PropTypes.func.isRequired,
};

export default MessageList;
