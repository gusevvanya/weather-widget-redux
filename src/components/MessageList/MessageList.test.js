import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import MessageList from './MessageList';

Enzyme.configure({ adapter: new Adapter() });

describe('MessageList', () => {
  let props;
  let messageListMount;
  let messageListShallow;

  beforeAll(() => {
    props = {
      onClearMessageClick: jest.fn(),
      messages: [{ value: 'New massage', type: 'success' }],
    };
  });

  beforeEach(() => {
    props.onClearMessageClick.mockClear();
    messageListMount = Enzyme.mount(<MessageList {...props} />);
    messageListShallow = Enzyme.shallow(<MessageList {...props} />);
  });

  it('should render', () => {
    expect(messageListShallow.debug()).toMatchSnapshot();
  });

  it('test buttonClickHandler method', () => {
    expect(props.onClearMessageClick.mock.calls.length).toBe(0);
    messageListMount.find('button').simulate('click');
    expect(props.onClearMessageClick.mock.calls.length).toBe(1);
  });

  it('render messageList if the param "messages" recived', () => {
    expect(messageListMount.find('.messageList').length).toBe(1);
    messageListMount.setProps({ messages: [] });
    expect(messageListMount.find('.messageList').length).toBe(0);
  });
});
