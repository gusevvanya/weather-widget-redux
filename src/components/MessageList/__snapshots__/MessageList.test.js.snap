// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList should render 1`] = `
"<div className=\\"messageList\\">
  <ul>
    <MessageItem value=\\"New massage\\" type=\\"success\\" />
  </ul>
  <button onClick={[Function]}>
    x
  </button>
</div>"
`;
